README.md

```diff
@@ -53,238 +55,332 @@ Attached to the release you will find two framework bundles.
+	...
+	config.urlDelegate = self;
+	[IterableAPI initializeWithApiKey:@"YOUR API KEY" launchOptions:launchOptions config:config];
+	...
+}
+	
+- (BOOL)handleIterableURL:(NSURL *)url context:(IterableActionContext *)context {
+	// Assuming you have a DeeplinkHandler class that handles all deep link URLs and navigates to the right place in the app
+	return [DeeplinkHandler handleUrl:url];
+}
+```
 		
-		Alternatively, call `getAndTrackDeeplink` along with a callback to handle the original deeplink url. You can use this method for any incoming URLs, as it will execute the callback without changing the URL for non-Iterable URLs.
+#### Handling email links
+	
+For universal links to work with link rewriting in emails, [set up an **apple-app-site-association** file](https://support.iterable.com/hc/articles/115000440206-Setting-up-iOS-Universal-Links) in the Iterable project. 
 
-		Swift:
-		
-		```swift
-		func application(_ application: UIApplication, continue userActivity: NSUserActivity, restorationHandler: @escaping ([Any]?) -> Void) -> Bool {
-			guard let url = userActivity.webpageURL else {
-				return false
-			}
-
-			IterableAPI.getAndTrack(deeplink: url) { (originalUrl) in
-				// Handle original url deeplink here
-			}
-			return true
-		}
-		```
-
-		Objective-C:
-		
-		```objective-c
-		- (BOOL)application:(UIApplication *)application
-				continueUserActivity(NSUserActivity *)userActivity 
-				restorationHandler:(void (^)(NSArray *restorableObjects))restorationHandler {
-
-			[IterableAPI getAndTrackDeeplink:iterableLink callbackBlock:^(NSString* originalURL) {
-				//Handle Original URL deeplink here
-			}];
-
-			return true;
-		}
-		```
+When an email link is clicked, your `UIApplicationDelegate`'s [application:continueUserActivity:restorationHandler:](https://developer.apple.com/documentation/uikit/uiapplicationdelegate/1623072-application?language=swift) method is called. If you already have an Iterable `urlDelegate` defined (see *Handling Links from Push Notifications* section above), the same handler can be used for email deep links by calling `handleUniversalLink:`.
+
+*Swift*
+	
+```swift
+func application(_ application: UIApplication, continue userActivity: NSUserActivity, restorationHandler: @escaping ([Any]?) -> Void) -> Bool {
+	guard let url = userActivity.webpageURL else {
+		return false
+	}
+
+	// This will track the click, retrieve the original URL and call `handleIterableURL:context:` with the original URL
+	return IterableAPI.handle(universalLink: url)
+}
+```
+
+*Objective-C*
+	
+```objective-c
+- (BOOL)application:(UIApplication *)application continueUserActivity(NSUserActivity *)userActivity restorationHandler:(void (^)(NSArray *restorableObjects))restorationHandler {
+	// This will track the click, retrieve the original URL and call `handleIterableURL:context:` with the original URL
+	return [IterableAPI handleUniversalLink:userActivity.webpageURL];
+}
+```
+	
+#### In-app notifications
```

Comment:
- In-app messages

Reply:
Thanks! In the future I'd love to help more with things like this, but I need to get spun up on using the SDK before I try to document how to do it :)